### DISCUSSION TITLE
how to override container **Entrypoint** and  **Cmd**

### DISCUSSION THREAD

**Discussion by vermapraveen** [testcontainers/testcontainers-dotnet#572] [Q&A] [answered]
I am trying to override **Entrypoint** and  **Cmd** of given docker image. While doing so: on CLI it works and container starts, I can see app logs, though when I try to construct in code using TestContainer, I see error. I am sure I am constructing it wrong, 

**_can someone suggest what I am doing wrong_** 

Works OK

> ```
> docker run 
> --env ENVIRONMENT=sit 
> --entrypoint /bin/sh harbor.myrepo.com/project1/events.handler:11684730-ae8b71bf 
> -c './integrationFolder/EventIntegration'
> ```

But for equivalent Code

> ```
>  private readonly TestcontainersContainer _eventHandlerContainer = 
>         new TestcontainersBuilder<TestcontainersContainer>()
> 	    .WithEntrypoint("/bin/sh")
> 	    .WithImage("harbor.myrepo.com/project1/events.handler:11684730-ae8b71bf")
> 	    .WithCommand("./integrationFolder/EventIntegration")
> 	    .WithEnvironment("ENVIRONMENT", "sit")
> 	    .Build();
> ```

I see below error

> ```
> Docker.DotNet.DockerApiException : Docker API responded with status code=BadRequest, response={
> "message":"failed to create shim task: OCI runtime create failed: runc create failed
> : unable to start container process: exec: \"./.csproj\": stat ./.csproj
> : no such file or directory: unknown"}
> ```

docker inspect gives below config in container
```
{
  "Config": {
    "Cmd": null,
    "WorkingDir": "/home/someuser",
    "Entrypoint": [
      "./.csproj"
    ]
  }
}
```

**Comment by HofmeisterAn** [comment, marked as answer]
That's not equivalent. You use `-c './integrationFolder/EventIntegration'` as `docker run` argument (even if I don't understand how that results in the shown `docker inspect` result). Try:

```csharp
.WithEntrypoint("/bin/sh", "-c")
.WithCommand("./integrationFolder/EventIntegration")
```
